runtime/parachains/src/scheduler.rs

```diff
@@ -375,7 +367,10 @@ impl<T: Config> Module<T> {
 	/// Schedule all unassigned cores, where possible. Provide a list of cores that should be considered
 	/// newly-freed along with the reason for them being freed. The list is assumed to be sorted in
 	/// ascending order by core index.
-	pub(crate) fn schedule(just_freed_cores: impl IntoIterator<Item = (CoreIndex, FreedReason)>) {
+	pub(crate) fn schedule(
```

Comment:
Need not to be a `pub(crate)`

In fact, it is not used anywhere except a single callsite in `clear_and_reschedule` and then in tests. I am wondering if we should inline this function into `clear_and_reschedule` (or the other way around) and use everywhere only one entrypoint.

Otherwise, I think it sucks a bit that we enter through `clear_and_reschedule` but test `schedule`.